x-pack/platform/plugins/shared/apm_sources_access/public/hooks.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import type { IHttpFetchError, ResponseErrorBody } from '@kbn/core-http-browser';
+import type { APIEndpoint, APIReturnType } from '../server';
+import { callSourcesAPI } from './api';
+
+function getDetailsFromErrorResponse(error: IHttpFetchError<ResponseErrorBody>) {
+  const message = error.body?.message ?? error.response?.statusText;
+  return (
+    <>
+      {message} ({error.response?.status})
+      <h5>
+        {i18n.translate('xpack.apmSourcesAccess.fetcher.error.url', {
+          defaultMessage: `URL`,
+        })}
+      </h5>
+      {error.response?.url}
+    </>
+  );
+}
+
+interface SourcesAPIFetcherParams<T extends APIEndpoint> {
+  pathname: T;
+  body?: unknown;
+}
+
+interface SourcesAPIReturn<T extends APIEndpoint> {
+  data: APIReturnType<T> | undefined;
+  refetch: () => void;
+}
+
+export const useSourcesAPIFetcher = <T extends APIEndpoint>({
```

Comment:
I wish there were something on platform level for this. There are many different flavours of this implementation

Reply:
Yeah, I decided to just tweak APM's `useFetcher` to pass an `AbortSignal` as a secondary param, so I can make use of it in the `apmSourcesAccess` APIs.